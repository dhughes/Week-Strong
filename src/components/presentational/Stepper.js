import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Minus, Plus } from './Icon';
import { Hbox } from './Box';
import theme from '../../util/theme';

const Container = styled(Hbox)`
  height: 3rem;
  line-height: 3rem;
  color: ${theme.primaryText};

  & > * {
    height: 3rem;
    width: 3rem;
  }

  & > h2{
    margin: 0;
    text-align: center;
    flex-grow: 1000;
  }

  & > svg {
    fill: currentColor;
  }
`;

const Stepper = props =>
  <Container>
    <Minus
      onClick={e =>
        props.onValueChange(props.value - props.step < props.minimum ? props.minimum : props.value - props.step)}
    />
    <h2>{props.value} {props.label}</h2>
    <Plus
      onClick={e =>
        props.onValueChange(props.value + props.step > props.maximum ? props.maximum : props.value + props.step)}
    />
  </Container>;

Stepper.defaultProps = {
  label: null,
  step: 1
};

Stepper.propTypes = {
  value: PropTypes.number.isRequired,
  step: PropTypes.number.isRequired,
  onValueChange: PropTypes.func.isRequired,
  minimum: PropTypes.number,
  maximum: PropTypes.number
};

export default Stepper;
