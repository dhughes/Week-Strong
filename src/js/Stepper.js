import React from 'react';
import { Minus, Plus } from './Icon';
import { Hbox } from './Box';
import styled from 'styled-components';

const Container = styled(Hbox)`
  height: 3em;
  line-height: 3em;
  color: ${props => props.theme.font};;

  & > * {
    height: 2em;
    width: 2em;
    margin: 0.2em;
    text-align: center;
    margin: 0px;
    padding: 0px;
  }

  & > h2{
    text-align: center;
    flex-grow: 1000;
  }

  & > svg {
    fill: currentColor;
  }

`;

const Stepper = props => {
  return (
    <Container>
      <Minus onClick={props.onStepDown} />
      <h2>{`${props.value} ${props.label}`.trim()}</h2>
      <Plus onClick={props.onStepUp} />
    </Container>
  );
};

Stepper.defaultProps = {
  value: 50,
  label: '',
  onStepUp: () => {},
  onStepDown: () => {}
};

export default Stepper;
