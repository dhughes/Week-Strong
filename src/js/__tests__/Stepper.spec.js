import React from 'react';
import { shallow, mount } from 'enzyme';
import Stepper from '../Stepper';
import sinon from 'sinon';

xtest('Stepper renders', () => {
  const component = shallow(<Stepper />);
  expect(component).toMatchSnapshot();
});

test('Stepper displays minus icon', () => {
  const component = shallow(<Stepper />);
  expect(component.find('Minus').length).toEqual(1);
});

test('Stepper displays plus icon', () => {
  const component = shallow(<Stepper />);
  expect(component.find('Plus').length).toEqual(1);
});

test('Stepper displays value', () => {
  const component = mount(<Stepper value="100" />);
  expect(component.text().indexOf('100')).toBeGreaterThan(-1);
});

test('Stepper displays label', () => {
  const component = mount(<Stepper value="100" label="widgets" />);
  expect(component.text().indexOf('widgets')).toBeGreaterThan(-1);
});

test('Stepper steps up', () => {
  const onButtonClick = sinon.spy();
  const component = shallow(<Stepper value="100" label="widgets" onStepUpClick={onButtonClick} />);

  // trigger click events
  component.find('Plus').simulate('click');
  expect(onButtonClick.getCalls().length).toBeGreaterThan(0);
});

test('Stepper steps down', () => {
  const onButtonClick = sinon.spy();
  const component = shallow(<Stepper value="100" label="widgets" onStepDownClick={onButtonClick} />);

  // trigger click events
  component.find('Minus').simulate('click');
  expect(onButtonClick.getCalls().length).toBeGreaterThan(0);
});
