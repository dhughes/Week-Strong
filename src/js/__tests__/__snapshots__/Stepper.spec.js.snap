// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper renders 1`] = `
<div>
  Stepper
</div>
`;
